module/Activity/src/Activity/Service/Activity.php

```diff
@@ -197,9 +222,54 @@ public function reset(ActivityModel $activity)
      */
     public function disapprove(ActivityModel $activity)
     {
+        $this->permissionManageActivityOrException();
+
         $activity->setStatus(ActivityModel::STATUS_DISAPPROVED);
         $em = $this->getServiceManager()->get('Doctrine\ORM\EntityManager');
         $em->persist($activity);
         $em->flush();
     }
+
+    /*
+     * Some general ACL functions
+     */
+
+    /**
+     * Checks if the user is allowed to view the activity, and if not, throws an used permission exception
+     */
+    public function permissionViewActivityOrException()
```

Comment:
It would be a bit better to make these permissions a little more granular. Like having separate `approve` and `reset` privileges, instead of a single privilege.